LDAP-primer.md

```diff
@@ -338,3 +361,26 @@ exist in the entry, so renaming an entry to a RDN that is not already present wo
 operations: one to add a new value and one to perform a renaming. To account for this situation,
 __modify DN__ will add the new naming attribute and value to the renamed entry automatically,
 and will also optionally delete the previous naming attribute-value pair.
+
+## 3.4. Extensions
+
+LDAP can be extended by servers in two main ways, using __controls__ and __extended operations__.
+There are lots of RFCs which standardize certain controls and extended operations.
+
+An extended operation is simply another operation, like "bind" or "search", that is not part of
+the core LDAP specifications. LDAP servers often support at least three extended operations:
+__StartTLS__, __WhoAmI__, and __PasswordModify__.
+
+Every LDAP operation is also allowed to contain a number of controls, which are there to change
+the operation's behaviour in some way. For example, the __Assertion__ control can be used with a
+"modify" operation to indicate that the server should only modify the entry if the contents of
+the control are true for the entry.
```

Comment:
Reformulated to provide a definition of a control and clarify what the contents of *Assertion* is.